core/src/mindustry/input/DesktopInput.java

```diff
@@ -306,7 +306,7 @@ public void update(){
                 cursorType = ui.unloadCursor;
             }
 
-            if(cursor.build != null && cursor.interactable(player.team()) && !isPlacing() && Math.abs(Core.input.axisTap(Binding.rotate)) > 0 && Core.input.keyDown(Binding.rotateplaced) && cursor.block().rotate){
+            if(cursor.build != null && cursor.interactable(player.team()) && !isPlacing() && Math.abs(Core.input.axisTap(Binding.rotate)) > 0 && Core.input.keyDown(Binding.rotateplaced) && cursor.block().rotate && cursor.block().quickRotate){
```

Comment:
This prevents rotation from happening, but it still draws the rotation selection over the block when you hold the rotate button.

Reply:
Fixed!